starlite/signature.py

```diff
@@ -194,7 +195,8 @@ def __init__(self, fn: "AnyCallable", plugins: List["PluginProtocol"], dependenc
         if fn is None:
             raise ImproperlyConfiguredException("Parameter `fn` to `SignatureModelFactory` cannot be `None`.")
         self.signature = Signature.from_callable(fn)
-        self.fn_name = fn.__name__ if hasattr(fn, "__name__") else "anonymous"
+        self.fn_name = getattr(fn, "__name__", "anonymous")
+        self.fn_module_name = getattr(fn, "__module__", "pydantic.main")
```

Comment:
Using `"pydantic.main"` as the default here preserves the current behaviour if there is no `__module__` attribute. We could similarly pass `__name__` so that `starlite.signature` is the module, but it's really just for the resolution to get access to global names so not sure that it matters a great deal.